android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -445,270 +449,301 @@ private MediaConstraints parseMediaConstraints(ReadableMap constraints) {
-                video = constraints.getMap("video");
-                if (video.hasKey("mandatory") && 
-                        video.getType("mandatory") == ReadableType.Map) {
-                    videoConstraintsManatory = video.getMap("mandatory");
-                }
-
-                // video resolution is mandatory
-                if (videoConstraintsManatory == null) {
-                    errorCallback.invoke(null, "video mandatory constraints not found");
-                    return;
-                }
-                
-                //videoConstraints = parseConstraints(video);
-                sourceId = getSourceIdConstraint(video);
-                facingMode
-                    = ReactBridgeUtil.getMapStrValue(video, "facingMode");
+                requestPermissions.add(PERMISSION_VIDEO);
                 break;
             default:
-                errorCallback.invoke(null, "invalid type of video constraints");
-                return;
+                break;
             }
+        }
+
+        // According to step 2 of the getUserMedia() algorithm,
+        // requestedMediaTypes is the set of media types in constraints with
+        // either a dictionary value or a value of "true".
+        // According to step 3 of the getUserMedia() algorithm, if
+        // requestedMediaTypes is the empty set, the method invocation fails
+        // with a TypeError.
+        if (requestPermissions.isEmpty()) {
+            errorCallback.invoke(
+                "TypeError",
+                "constraints requests no media types");
+            return;
+        }
 
-            if (enableVideo) {
-                Log.i(TAG, "getUserMedia(video): video: " + video
-                        + ", videoConstraintsManatory: " + videoConstraintsManatory
-                        + ", sourceId: " + sourceId);
-
-                Context context = (Context) getReactApplicationContext();
-                final boolean isFacing = (facingMode != null && facingMode.equals("environment"))
-                    ? false : true;
-                VideoCapturer videoCapturer = null;
-
-                // NOTE: to support Camera2, the device should:
-                //   1. Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP
-                //   2. all camera support level should greater than LEGACY
-                //   see: https://developer.android.com/reference/android/hardware/camera2/CameraCharacteristics.html#INFO_SUPPORTED_HARDWARE_LEVEL
-                // TODO Enable camera2 enumerator
-                if (false && Camera2Enumerator.isSupported(context)) {
-                    Log.d(TAG, "Creating video capturer using Camera2 API.");
-                    videoCapturer = createVideoCapturer(
-                        new Camera2Enumerator(context), isFacing, sourceId);
+        final PermissionUtils.Callback callback = new PermissionUtils.Callback() {
+            @Override
+            public void invoke(String permission, int grantResult) {
```

Comment:
I handled that differently, this was checked further down, after all permissions were checked. Then the logic worked out if we had the ones that were needed vs the ones that were requested.